kani-compiler/src/kani_middle/transform/body.rs

```diff
@@ -35,6 +36,16 @@ pub struct MutableBody {
 
     /// The span that covers the entire function body.
     span: Span,
+
+    /// Set of basic block indices for which analyzing first statement should be skipped.
```

Comment:
Previously, all checks were inserted before the source instruction, which was in turn moved to the next basic block. Hence, we would not need to look at it again as a part of new basic block. Now, if the check is inserted after the source instruction, we still need to look at the first statement of the new basic block, so we need to keep track of which basic blocks were created as a part of injecting checks after the source instruction.

Reply:
I am not convinced this should be part of the `MutableBody` though. Isn't this a transformation pass responsibility?

Reply:
I was under the impression that depending on the basic block splitting logic, it would be hard to know ahead whether we need to skip the new block. But it seems to me that it's actually not the case, I will move that logic into the transformation pass.